bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/connection/Clip2Bridge.java

```diff
@@ -369,15 +404,50 @@ public boolean onIdleTimeout(@Nullable Session session) {
 
         @Override
         public void onPing(@Nullable Session session, @Nullable PingFrame frame) {
-            checkAliveOk();
-            if (Objects.nonNull(session) && Objects.nonNull(frame) && !frame.isReply()) {
-                session.ping(new PingFrame(true), Callback.NOOP);
+            Objects.requireNonNull(session);
+            Objects.requireNonNull(frame);
+            if (http2Session == session) {
+                checkAliveOk();
+                if (!frame.isReply()) {
+                    session.ping(new PingFrame(true), Callback.NOOP);
+                }
             }
         }
 
         @Override
         public void onReset(@Nullable Session session, @Nullable ResetFrame frame) {
-            fatalErrorDelayed(this, new Http2Exception(Http2Error.RESET));
+            Objects.requireNonNull(session);
+            fatalErrorDelayed(this, new Http2Exception(Http2Error.RESET), session.hashCode(), -1);
+        }
+    }
+
+    /**
+     * Synchronizer for accessing the HTTP2 session object. This method wraps the 'sessionUseCreateLock' ReadWriteLock
+     * so that GET/PUT methods can access the session on multiple concurrent threads via the 'read' access lock, yet are
+     * forced to wait if the session is being created via its single thread access 'write' lock.
+     */
+    private class SessionSynchronizer implements AutoCloseable {
+        private final boolean exclusive;
+        private final boolean locked;
+
+        SessionSynchronizer(boolean requireExclusiveAccess) throws InterruptedException {
+            exclusive = requireExclusiveAccess;
+            if (exclusive) {
+                locked = sessionUseCreateLock.writeLock().tryLock(TIMEOUT_SECONDS, TimeUnit.SECONDS);
+            } else {
+                locked = sessionUseCreateLock.readLock().tryLock(TIMEOUT_SECONDS, TimeUnit.SECONDS);
+            }
```

Comment:
```
Lock lock = exclusive ? sessionUseCreateLock.writeLock() : sessionUseCreateLock.readLock();
if (lock.tryLock()) {
    this.lock = lock;
}
```

... and then in `close()` unlock only if there's a lock. No other members are needed with that approach.

Reply:
Based on the ReentrantReadWritock JavaDoc the tryLock() method is less predictable.

Reply:
^
```
/**
         * Acquires the read lock if the write lock is not held by
         * another thread within the given waiting time and the
         * current thread has not been {@linkplain Thread#interrupt
         * interrupted}.
         *
         * <p>Acquires the read lock if the write lock is not held by
         * another thread and returns immediately with the value
         * {@code true}. If this lock has been set to use a fair
         * ordering policy then an available lock <em>will not</em> be
         * acquired if any other threads are waiting for the
         * lock. This is in contrast to the {@link #tryLock()}
         * method. If you want a timed {@code tryLock} that does
         * permit barging on a fair lock then combine the timed and
         * un-timed forms together:
         *
         * <pre> {@code
         * if (lock.tryLock() ||
         *     lock.tryLock(timeout, unit)) {
         *   ...
         * }}</pre>
         *
         * <p>If the write lock is held by another thread then the
         * current thread becomes disabled for thread scheduling
         * purposes and lies dormant until one of three things happens:
         *
         * <ul>
         *
         * <li>The read lock is acquired by the current thread; or
         *
         * <li>Some other thread {@linkplain Thread#interrupt interrupts}
         * the current thread; or
         *
         * <li>The specified waiting time elapses.
         *
         * </ul>
         *
         * <p>If the read lock is acquired then the value {@code true} is
         * returned.
         *
         * <p>If the current thread:
         *
         * <ul>
         *
         * <li>has its interrupted status set on entry to this method; or
         *
         * <li>is {@linkplain Thread#interrupt interrupted} while
         * acquiring the read lock,
         *
         * </ul> then {@link InterruptedException} is thrown and the
         * current thread's interrupted status is cleared.
         *
         * <p>If the specified waiting time elapses then the value
         * {@code false} is returned.  If the time is less than or
         * equal to zero, the method will not wait at all.
         *
         * <p>In this implementation, as this method is an explicit
         * interruption point, preference is given to responding to
         * the interrupt over normal or reentrant acquisition of the
         * lock, and over reporting the elapse of the waiting time.
         *
         * @param timeout the time to wait for the read lock
         * @param unit the time unit of the timeout argument
         * @return {@code true} if the read lock was acquired
         * @throws InterruptedException if the current thread is interrupted
         * @throws NullPointerException if the time unit is null
         */
```

Reply:
Oh. I get what you are saying, now.

Reply:
Ok. Done.